packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -189,8 +188,11 @@ function s(
         !isUndefined(slotset.slotAssignments[slotName]) &&
         slotset.slotAssignments[slotName].length !== 0
     ) {
-        children = slotset.slotAssignments[slotName].reduce((accumulator, vnode) => {
-            if (vnode) {
+        const newChildren: VNode[] = [];
+        const slotAssignments = slotset.slotAssignments[slotName];
+        for (let i = 0; i < slotAssignments.length; i++) {
```

Comment:
I didn't check what the fastest way to iterate is (e.g. compared to `for (const vnode of slotAssignments) {}`. But we do use this pattern in lots of places in the code.